### ISSUE TITLE
Make pod_name and pod_namespace optional

### ISSUE THREAD

**Issue by mneverov** [kubernetes-sigs/node-ipam-controller#39]
Currently, [POD_NAME and POD_NAMESPACE](https://github.com/kubernetes-sigs/node-ipam-controller/blob/0a1e0fdb6eb8f0308b52e84fddff9f322334c077/pkg/leaderelection/leaderelection.go#L26-L33) environment variables are mandatory. As a user I don't want to specify them but just run the controller.
Investigate if 
 - we need to use resource lock
 - it is possible to deduct name/namespace from os.host or by other means

Update readme with both variables, descriptions and default values from the investigation.

**Comment by aojea** [comment]
actually those need to be passed as parameters https://github.com/kubernetes-sigs/node-ipam-controller/blob/0a1e0fdb6eb8f0308b52e84fddff9f322334c077/pkg/leaderelection/leaderelection.go#L16-L22

you can use some defaults , for namespace "default, and for id is commont to use os.Hostname()

**Comment by mneverov** [comment]
> actually those need to be passed as parameters

The biggest problem at the moment that they are not [specified](https://github.com/kubernetes-sigs/node-ipam-controller?tab=readme-ov-file#config) and by default the controller runs in leader election mode, but without these variables it fails to start. Would be nice to address it

**Comment by aojea** [comment]
yeah, we can define some default values , pod_name = os.Hostname() and pod_namespace = "default" or "kube-system", whatever we use to install our controller, I can not remember now

**Comment by mneverov** [comment]
maybe also can be used staging/src/k8s.io/component-helpers/node/util/hostname.go:28

**Comment by aojea** [comment]
yeah, that works too
